pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,334 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+
+import re
+import time
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+from pymeasure.instruments.validators import strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class EurotestHPP120256(Instrument):
+    """ Represents the Euro Test High Voltage DC Source model HPP-120-256
+    and provides a high-level for interacting with the instrument.
+
+    .. code-block:: python
+
+    hpp120256 = EurotestHPP120256("GPIB0::20::INSTR")
+
+    response = hpp120256.id
+    print(response)
+    status_lam = hpp120256.lam_status()
+    print(status_lam)
+    status = hpp120256.status()
+    print(status)
+
+    hpp120256.ramp_to_zero(100.0)
+
+    hpp120256.voltage_ramp = 50.0  # V/s
+    time.sleep(command_delay)
+    hpp120256.current_limit = 2.0  # mA
+    time.sleep(command_delay)
+    inst.enable_kill = "ON"  # Enable over-current protection
+    time.sleep(command_delay)
+    inst.enable_output = "ON"
+    time.sleep(1.0)  # Give time to output on
+
+    hpp120256.wait_for_voltage_output_set(1.0, 40.0)
+
+    # Here voltage HV output should be at 0.0 kV
+
+    print("Setting the output voltage to 1.0kV...")
+    hpp120256.voltage = 1.0  # kV
+
+    # Now HV output should be rising to reach the 1.0kV at 50.0 V/s
+
+    hpp120256.wait_for_voltage_output_set(1.0, 40.0)
+
+    # Here voltage HV output should be at 1.0 kV
+
+    hpp120256.shutdown(100.0)
+
+    hpp120256.wait_for_voltage_output_set(1.0, 60.0)
+
+    # Here voltage HV output should be at 0.0 kV
+
+    # Now the HV voltage source is in safe state
+
+    """
+
+    VOLTAGE_RANGE = [0.0, 12.0]  # kVolts
+    CURRENT_RANGE = [0.0, 25.0]  # mAmps
+    VOLTAGE_RAMP_RANGE = [10, 3000]  # V/s
+    COMMAND_DELAY = 0.2  # s
+
+    # ####################################
+    # # EuroTest-Command set. Non SCPI commands.
+    # ####################################
+
+    voltage = Instrument.control(
+        "STATUS,U", "U,%.3fkV",
+        """ A floating point property that represents the output voltage
+        setting (in kV) of the HV Source in kVolts. This property can be set.
+        When this property acts as get, the instrument will return a string like this:
+        U, RANGE=3.000kV, VALUE=2.458kV, then voltage will return a tuple
+        2.458 hence, the convenience of the get_process.""",
+        validator=strict_range,
+        values=VOLTAGE_RANGE,
+        get_process=lambda v: float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
```

Comment:
why don't you look for `r'VALUE=([+-]?[\d\.]+)'`? That way, you get the right part and the regex is smaller and more concise.

Btw, looking at that expression, you do not include the plus/minus in the float, such that `"VALUES=-123"` returns `123` instead of `-123`.

Reply:
Yes. Pretty good. I will use it. But not in such fashion. Below my explanation:
What I found when the instrument response, for example, to voltage is an array of strings not a string like this "U, RANGE=3.000kV, VALUE=2.458kV". The instrument response this list ["U", "RANGE=3.000kV", "VALUE=2.458kV"]. Thats the reason I use re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2]) over v[2]. 
I will use re.findall(r'([+-]?[\d\.]+)', v[2]) instead.
Also I will rewrite the docstring replacing "U, RANGE=3.000kV, VALUE=2.458kV" for ["U", "RANGE=3.000kV", "VALUE=2.458kV"]

Note: Sincerely I'm not sure why I obtain a list ( ["U", "RANGE=3.000kV", "VALUE=2.458kV"]) instead of an string like this "U, RANGE=3.000kV, VALUE=2.458kV". Perhaps it is an internal transformation of pymeasure? What do you think?

Many thanks

Reply:
The pymeasure properties call `Instrument.values`, which does splitting the string into several (in order to read several values), default is at ",". 
If you give it `separator=None` (also applicable for properties), it won't split at all.

Or you might use it for your own advantage.